### PR TITLE
Add `purge-db-safe` flag which requires manual confirmation

### PR DESCRIPTION
**Pull request by macladson**
## Issue Addressed

https://github.com/sigp/lighthouse/issues/5028

## Proposed Changes

Add a new flag called `--purge-db-safe` which functions like `--purge-db` but requires manual confirmation. As such it is not suitable for environments that are designed to be automated.

## Additional Info

I moved the `stdin-inputs` flag to be a global flag under the `lighthouse` command, as I felt this is a cleaner solution than currently, even if not every subcommand will use it. Happy to change it back if someone has a different opinion.

### CODE REVIEW

**Comment by dapplion** [comment]
What about waiting for 5.0.0 and sliding a breaking change to make `--purge-db` require confirmation?

**Comment by macladson** [comment]
> What about waiting for 5.0.0 and sliding a breaking change to make `--purge-db` require confirmation?

Yeah happy to do that. I do think we should have two flags regardless, so automated setups have an option.
Something like `--purge-db-auto`? It's definitely more natural for `--purge-db` to be the safe one. Depends if the breaking change is worth it though.

File: beacon_node/src/config.rs [hunk 1]
```diff
@@ -48,26 +51,33 @@ pub fn get_config<E: EthSpec>(
...
+            let confirmation = read_input_from_user(stdin_inputs)?;
+
+            if confirmation == PURGE_DB_CONFIRMATION {
+                eprintln!("Database was deleted.");
+                let chain_db = client_config.get_db_path();
+                let freezer_db = client_config.get_freezer_db_path();
+                let blobs_db = client_config.get_blobs_db_path();
+                purge_db(chain_db, freezer_db, blobs_db)?;
+            } else {
+                eprintln!("Database was not deleted.");
```
Comment (dapplion) [inline beacon_node/src/config.rs:79]: Should you error exit here instead of ignoring the request to purge the DB?

**Comment by macladson** [comment]
Closed in favour of #6154

**Pull request closed without merging**